zipkin-server/src/main/java/zipkin2/server/internal/eureka/ZipkinEurekaDiscoveryProperties.java

```diff
@@ -30,21 +33,45 @@
  */
 @ConfigurationProperties("zipkin.discovery.eureka")
 class ZipkinEurekaDiscoveryProperties implements Serializable { // for Spark jobs
-  /** URL of the Eureka v2 endpoint. e.g. http://eureka:8761/eureka/v2 */
-  private String serviceUrl;
+  /**
+   * URL of the Eureka v2 endpoint. e.g. http://eureka:8761/eureka/v2
+   *
+   * <p>Note: When present, {@link URI#getUserInfo() userInfo} credentials will be used for BASIC
+   * authentication. For example, if the URL is "https://myuser:mypassword@1.1.3.1/eureka/v2/",
+   * requests to Eureka will authenticate with the user "myuser" and password "mypassword".
+   */
+  private URI serviceUrl;
   /** The appName property of this instance of zipkin. */
   private String appName;
   /** The instanceId property of this instance of zipkin. */
   private String instanceId;
   /** The hostName property of this instance of zipkin. */
   private String hostname;
 
-  public String getServiceUrl() {
+  private BasicToken auth;
+
+  public URI getServiceUrl() {
     return serviceUrl;
   }
 
-  public void setServiceUrl(String serviceUrl) {
-    this.serviceUrl = emptyToNull(serviceUrl);
+  public void setServiceUrl(URI serviceUrl) {
+    if (serviceUrl.getUserInfo() != null) {
+      String[] ui = serviceUrl.getUserInfo().split(":");
+      if (ui.length == 2) {
+        auth = BasicToken.ofBasic(ui[0], ui[1]);
+      }
+    }
+    this.serviceUrl = stripBaseUrl(serviceUrl);
+  }
+
+  // Strip the credentials and any invalid query or fragment from the URI
+  static URI stripBaseUrl(URI serviceUrl) {
+    try {
+      return new URI(serviceUrl.getScheme(), null, serviceUrl.getHost(), serviceUrl.getPort(),
```

Comment:
Curious why we the query fragment are considered invalid?

Reply:
because there are no global query params or fragments defined in eureka's api. I can reference that, good idea!

Reply:
ps I actually didn't mean to put this class in the PR, so will remove it ;) this was for a follow-up lol